counsel.el

```diff
@@ -1855,11 +1859,47 @@ When INITIAL-INPUT is non-nil, use it in the minibuffer during completion."
   "Expand CMD that ends in \"ls\" with switches."
   (concat cmd " " counsel-dired-listing-switches " | sed -e \"s/^/  /\""))
 
+(defvar counsel-file-name-filter-alist
+  '(("ag -i '%s'" . t)
+    ("ack -i '%s'" . t)
+    ("perl -ne '/(%s.*)/i && print \"$1\\n\";'" . t)
+    ("grep -i -E '%s'"))
+  "Alist of file name filtering commands.
+The car is a shell command and the cdr is t when the shell
+command supports look-arounds. The executable for the commands
+will be checked for existence via `executable-find'. The first
+one that exists will be used.")
+
+(defun counsel--file-name-filter (&optional use-ignore)
+  "Return a command that filters a file list to match ivy candidates.
+If USE-IGNORE is non-nil, try to generate a command that respects
+`counsel-find-file-ignore-regexp'."
+  (let ((regex ivy--old-re)
+        (ignore-re (list (counsel--elisp-to-pcre
+                          counsel-find-file-ignore-regexp)))
+        (cmd (cl-find-if
+              (lambda (x)
+                (executable-find
+                 (car (split-string (car x)))))
+              counsel-file-name-filter-alist)))
+    (and use-ignore ivy-use-ignore
+         counsel-find-file-ignore-regexp
+         (cdr cmd)
+         (not (string-match-p "\\`\\." ivy-text))
```

Comment:
Just checking: is this meant to match a full stop at the start of `ivy-text`, or is it supposed to match `(rx bos)` as well?

Reply:
Only match a full stop. See the equivalent code in `counsel--file-name-filter`
 https://github.com/abo-abo/swiper/blob/0c8cfe80ef251161a703014bee24c070d4049f27/counsel.el#L1793